cmd/acra-migrate-keys/migrate-keys/command-line.go

```diff
@@ -70,35 +78,54 @@ var Params CommandLineParams
 // RegisterCommandLineParams registers command-line options for parsing.
 func RegisterCommandLineParams() *CommandLineParams {
 	// Source key store
-	flag.StringVar(&Params.Src.KeyStoreVersion, "src_keystore", "v1", "key store format to use: v1 (current), v2 (new)")
+	flag.StringVar(&Params.Src.KeyStoreVersion, "src_keystore", "", "key store format to use: v1 (current), v2 (new)")
 	flag.StringVar(&Params.Src.KeyDir, "src_keys_dir", defaultSrcDir, "path to source key directory")
 	flag.StringVar(&Params.Src.KeyDirPublic, "src_keys_dir_public", defaultSrcDir, "path to source key directory for public keys")
 	// Destination key store
-	flag.StringVar(&Params.Dst.KeyStoreVersion, "dst_keystore", "v2", "key store format to use: v1 (current), v2 (new)")
+	flag.StringVar(&Params.Dst.KeyStoreVersion, "dst_keystore", "", "key store format to use: v1 (current), v2 (new)")
 	flag.StringVar(&Params.Dst.KeyDir, "dst_keys_dir", "", "path to destination key directory (default \".acrakeys.migrated\")")
 	flag.StringVar(&Params.Dst.KeyDirPublic, "dst_keys_dir_public", "", "path to destination key directory for public keys (default \".acrakeys.migrated\")")
 	// Miscellaneous
 	flag.BoolVar(&Params.Misc.DryRun, "dry_run", false, "try migration without writing to the output key store")
 	flag.BoolVar(&Params.Misc.Force, "force", false, "write to output key store even if it exists")
-	flag.BoolVar(&Params.Misc.LogDebug, "d", false, "log debug messages to stderr")
-	flag.BoolVar(&Params.Misc.LogVerbose, "v", false, "log everything to stderr")
+	flag.BoolVar(&Params.Misc.logDebug, "d", false, "log debug messages to stderr")
+	flag.BoolVar(&Params.Misc.logVerbose, "v", false, "log everything to stderr")
+	Params.Misc.LogLevel = log.GetLevel()
 
 	return &Params
 }
 
-// SetDefaults initializes default paramater values.
-func (params *CommandLineParams) SetDefaults() {
-	if params.Misc.LogDebug {
-		log.SetLevel(log.DebugLevel)
+// Parse parses command-line, validates commad-line options, and initializes default paramater values.
+func (params *CommandLineParams) Parse() error {
+	err := cmd.Parse(defaultConfigPath, serviceName)
+	if err != nil {
+		return err
+	}
+
+	if params.Src.KeyStoreVersion == "" {
+		log.Warning("Missing required argument: --src_keystore={v1|v2}")
 	}
-	if params.Misc.LogVerbose {
-		log.SetLevel(log.TraceLevel)
+	if params.Dst.KeyStoreVersion == "" {
+		log.Warning("Missing required argument: --dst_keystore={v1|v2}")
+	}
+	if params.Src.KeyStoreVersion == "" || params.Dst.KeyStoreVersion == "" {
+		return ErrMissingFormat
 	}
 
+	if params.Misc.logDebug {
+		params.Misc.LogLevel = log.DebugLevel
+	}
+	if params.Misc.logVerbose {
+		params.Misc.LogLevel = log.TraceLevel
```

Comment:
in other our tools we set InfoLevel for `-v` flag and `DebugLevel` for `-d` flag. In most cases and other tools debug level provide more information than "verbose", because debug level it's for developers and verbose for end users. In logrus TraceLevel is most verbose level. If we want to provide turn on TraceLevel for developers then maybe do it by `-d` flag?

p.s. plus when I first looked on this PR, I thought to change debug+verbose flags to one "level" flag because thought that MiscParams and these flags will be used in code with some if/else branches. And only now when you provided both versions I see, that we always use -d/-v and these flags in `MiscParams` used once only when configure logging at start. And it's okay previous version )

Reply:
Hm... Right... I've just reviewed the tools: as for log levels, `-v` enables Info logs, and `-d` enables Debug logs. This is typically performed only after the initial startup, so by default only Warning and higher are printed.

In the context of `acra-migrate-keys`, Info is used for regular output of status messages. So it seems I thought that mapping Trace to verbose makes more sense. The trace level is much more verbose as key store implementations starts spewing out lots of traces, while on Debug level normally there are only a couple of status reports and more detailed information about errors (if they arise).

However, it is true that the verbose level does not make much sense to users, it's only useful for developers and debugging.

Well, it seems to be harmless so I guess we can leave it as is. Any objections? @shadinua?

Reply:
imho, we have 2 ways. change level usage in code (trace -> debug, debug -> info) to be consistent with other code (I don't remember where we use TraceLevel in other our components), or just change mapping of flags to logLeve and configure TraceLevel for `-d` flag and DebugLevel for `-v` flag, to be consistent with other our tools by cli.

Reply:
Alrighty. I've reverted the change which exposes logrus log level after parameters parsing (not really useful since no one is currently interested in that option). I have also swapped the `-d` and `-v` flags. Now `-v` is somewhat useful to users (outputs Debug messages which can be used to diagnose key store access errors). The `-d` flag goes full verbose on all minor tasks that the key store does: open this key, try decrypting that, process that operation, etc.